pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+}
+
+type ClusterDNSConfig struct {
+  // clusterDNS is a list of IP address for the cluster DNS server.  If set,
+  // kubelet will configure all containers to use this for DNS resolution
+  // instead of the host's DNS servers
+  ClusterDNS []string `json:"clusterDNS"`
+  // clusterDomain is the DNS domain for this cluster. If set, kubelet will
+  // configure all containers to search this domain in addition to the
+  // host's search domains.
+  ClusterDomain string `json:"clusterDomain"`
+}
+
+type KubeAPIConfig struct {
+  /* requests to API server */
+  // contentType is contentType of requests sent to apiserver.
+  ContentType string `json:"contentType"`
+  // kubeAPIQPS is the QPS to use while talking with kubernetes apiserver
+  KubeAPIQPS *int32 `json:"kubeAPIQPS"`
+  // kubeAPIBurst is the burst to allow while talking with kubernetes
+  // apiserver
+  KubeAPIBurst int32 `json:"kubeAPIBurst"`
+  // syncFrequency is the max period between synchronizing running
+  // containers and config
+  SyncFrequency metav1.Duration `json:"syncFrequency"` // TODO(kc-refactor): This feels right here but take another look later.
+}
+
+type FeatureGates struct {
+  // featureGates is a string of comma-separated key=value pairs that describe feature
+  // gates for alpha/experimental features.
+  FeatureGates string `json:"featureGates,omitempty"`
+}
+
+// Management of Node resources
+type NodeResourcesConfig struct {
+  // oomScoreAdj is The oom-score-adj value for kubelet process. Values
+  // must be within the range [-1000, 1000].
+  OOMScoreAdj *int32 `json:"oomScoreAdj"`
+  // A set of ResourceName=Percentage (e.g. memory=50%) pairs that describe
+  // how pod resource requests are reserved at the QoS level.
+  // Currently only memory is supported. [default=none]"
+  ExperimentalQOSReserved map[string]string `json:"experimentalQOSReserved"`
+  // maxPods is the number of pods that can run on this Kubelet.
+  MaxPods int32 `json:"maxPods"`
+  // Maximum number of pods per core. Cannot exceed MaxPods
+  PodsPerCore int32 `json:"podsPerCore"`
+  // cpuCFSQuota is Enable CPU CFS quota enforcement for containers that
+  // specify CPU limits
+  CPUCFSQuota *bool `json:"cpuCFSQuota"`
+}
+
+type CgroupsConfig struct {
+  // kubeletCgroups is the absolute name of cgroups to isolate the kubelet in.
+  KubeletCgroups string `json:"kubeletCgroups"`
+  // systemCgroups is absolute name of cgroups in which to place
+  // all non-kernel processes that are not already in a container. Empty
+  // for no container. Rolling back the flag requires a reboot.
+  SystemCgroups string `json:"systemCgroups"`
+  // runtimeCgroups are cgroups that container runtime is expected to be isolated in.
+  RuntimeCgroups string `json:"runtimeCgroups"` // TODO(kc-refactor): Does this become part of CRI now?
+  // cgroupRoot is the root cgroup to use for pods. This is handled by the
+  // container runtime on a best effort basis.
+  CgroupRoot string `json:"cgroupRoot"`
```

Comment:
i thought we updated this text.  with cgroups per qos enabled, this is the cgroup that will be the parent of kubepods cgroup for all end-user pods.